pkg/assembler/graphdb/neo4j/graphdb.go

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright 2022 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Note: All this code here is temporary and will change often. This module
+// must be a leaf in the dependency tree!
+
+package neo4j
+
+import (
+	"github.com/neo4j/neo4j-go-driver/v4/neo4j"
+)
+
+// AuthToken is the authentication token needed for connecting to the graph
+// database. Use the `CreateAuthToken...` functions to create a token.
+type AuthToken = neo4j.AuthToken
+
+// CreateAuthTokenWithUsernameAndPassword creates a simple authentication token
+// with username, password and authentication realm. This is the method to call
+// in most scenarios when you need an `AuthToken`.
+func CreateAuthTokenWithUsernameAndPassword(username string, password string, realm string) AuthToken {
+	return neo4j.BasicAuth(username, password, realm)
+}
+
+// Neo4jClient represents a client to the graph database.
+// TODO(mihaimaruseac): Switch to v5 and `...WithContext` API when v5 is released.
+type Neo4jClient interface {
```

Comment:
Maybe its my lack of understanding on this PR but shouldn't the interface be defined at like graphdb.go and neo4J be an implementation of that interface? So that we can have a pluggable nature with more graphDB integration in the future?

Reply:
I agree, but which `graphdb.go` are you referring to? Are you referring to the one in the `neo4j` package, or the one in the `graphdb` package?